### PR TITLE
fix(curriculum): cat photo app workshop step 16-20

### LINKED ISSUE #56312: Update steps 16-20 of cat photo workshop to use specific assert methods where appropriate
We recently updated the guidelines for writing assertions here 
https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#writing-tests

For consistency and matching what is in the docs, we should slowly start to update the tests for the workshops and labs in the new frontend cert.

For this issue, just focus on updating the tests for steps 16-20 of the cat photo app workshop.
We don't want to have a large PR that updates a whole bunch of tests in a bunch of steps and makes it harder to review. 

Here is the link to the workshop:
https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app

The tests should be updated, where appropriate, to use specific methods like `assert.equal`, `assert.exists`, `assert.match`, etc instead of just `assert`.

Here is the complete API reference 
https://www.chaijs.com/api/assert/#method_exists

Just focus on the new frontend cert and don't worry about updating the RWD cat photo app since that will become legacy anyway.

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56312

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984218720e29393b1adad2.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ const childrenOfSection = [...document.querySelector('section').childNodes];
 const foundElements = childrenOfSection.filter((child) => {
   return ['H2', 'A', 'P'].includes(child.nodeName);
 });
-assert(foundElements.length === 3);
+assert.equal(foundElements.length, 3);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984218720e29393b1adad2.md:50]: `assert.lengthOf` is more appropriate

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 2]
```diff
@@ -14,25 +14,25 @@ It is time to add a new section. Add a second `section` element below the existi
 Your `section` element should have an opening tag. Opening tags have this syntax: `<elementName>`.
 
 ```js
-assert(document.querySelectorAll('section').length >= 2);
+assert.isTrue(document.querySelectorAll('section').length >= 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:17]: `assert.isAtLeast`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 3]
```diff
@@ -14,25 +14,25 @@ It is time to add a new section. Add a second `section` element below the existi
...
 ```
 
 You should only add one opening `section` tag. Please remove any extras.
 
 ```js
-assert(document.querySelectorAll('section').length === 2);
+assert.equal(document.querySelectorAll('section').length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:23]: `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 4]
```diff
@@ -14,25 +14,25 @@ It is time to add a new section. Add a second `section` element below the existi
...
 ```
 
 Your `section` element should have a closing tag. Closing tags have a `/` just after the `<` character.
 
 ```js
-assert(code.match(/<\/section>/g).length >= 2);
+assert.isTrue(code.match(/<\/section>/g).length >= 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:29]: `assert.isAtLeast`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 5]
```diff
@@ -14,25 +14,25 @@ It is time to add a new section. Add a second `section` element below the existi
...
 ```
 
 You should only add one closing `section` tag. Please remove any extras.
 
 ```js
-assert(code.match(/<\/section>/g).length === 2);
+assert.equal(code.match(/<\/section>/g).length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:35]: `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 6]
```diff
@@ -44,7 +44,7 @@ const childrenOf1stSection = [
 const foundElems = childrenOf1stSection.filter((child) => {
   return child.nodeName === 'SECTION';
 });
-assert(foundElems.length === 0);
+assert.equal(foundElems.length, 0);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:47]: `assert.isEmpty`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 7]
```diff
@@ -54,7 +54,7 @@ const childrenOfMain = [...document.querySelector('main').children];
 const foundElems = childrenOfMain.filter((child) => {
   return child.nodeName === 'SECTION';
 });
-assert(foundElems.length === 2);
+assert.equal(foundElems.length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:57]: `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842cee55b683aa13ee758.md [hunk 8]
```diff
@@ -14,7 +14,7 @@ Within the second `section` element, add a new `h2` element with the text `Cat L
 Your `section` element should have an opening tag. Opening tags have this syntax: `<elementName>`.
 
 ```js
-assert(
+assert.isTrue(
   document.querySelectorAll('section').length === 2 &&
     code.match(/<\/section>/g).length === 2
 );
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842cee55b683aa13ee758.md:20]: two asserts with `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842cee55b683aa13ee758.md [hunk 9]
```diff
@@ -23,26 +23,26 @@ assert(
 Your `h2` element should have an opening tag. Opening tags have this syntax: `<elementName>`.
 
 ```js
-assert(document.querySelectorAll('h2').length === 2);
+assert.equal(document.querySelectorAll('h2').length, 2);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842cee55b683aa13ee758.md:26]: `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842cee55b683aa13ee758.md [hunk 10]
```diff
@@ -23,26 +23,26 @@ assert(
...
 const secondSection = document.querySelectorAll('section')[1];
-assert(secondSection.lastElementChild.nodeName === 'H2');
+assert.equal(secondSection.lastElementChild.nodeName, 'H2');
 ```
 
 The second `h2` element should have the text `Cat Lists`. You have either omitted the text or have a typo.
 
 ```js
-assert(
+assert.isTrue(
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842cee55b683aa13ee758.md:45]: `assert.equal`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842f261665d3b4a42df81.md [hunk 11]
```diff
@@ -18,7 +18,7 @@ After the last `h2` element of the second `section` element, add an `h3` element
 The second `section` element appears to be missing or does not have both an opening and closing tag.
 
 ```js
-assert(
+assert.isTrue(
   document.querySelectorAll('main > section')[1] &&
     code.match(/\<\/section>/g).length == 2
 );
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842f261665d3b4a42df81.md:24]: first `assert.exists`, then `assert.lengthOf`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842f261665d3b4a42df81.md [hunk 12]
```diff
@@ -27,16 +27,16 @@ assert(
...
+  document.querySelectorAll('main > section')[1].lastElementChild.nodeName,
     'H3'
 );
 ```
 
 The `h3` element right above the second `section` element's closing tag should have the text `Things cats love:`. Make sure to include the colon at the end of the text.
 
 ```js
-assert(
+assert.isTrue(
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842f261665d3b4a42df81.md:39]: `assert.equal`

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842f261665d3b4a42df81.md [hunk 13]
```diff
@@ -49,7 +49,7 @@ There should be an `h2` element with the text `Cat Lists` above the last `h3` el
 ```js
 const secondSectionLastElemNode = document.querySelectorAll('main > section')[1]
   .lastElementChild;
-assert(
+assert.isTrue(
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/669842f261665d3b4a42df81.md:52]: `assert.equal`

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
the title of the workshop is cat photo app, please change to that so that it is understandable for what workshop this issue is for

**Comment by gagan-bhullar-tech** [comment]
@ilenia-magoni updated PR

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984218720e29393b1adad2.md [hunk 14]
```diff
@@ -47,7 +47,7 @@ const childrenOfSection = [...document.querySelector('section').childNodes];
 const foundElements = childrenOfSection.filter((child) => {
   return ['H2', 'A', 'P'].includes(child.nodeName);
 });
-assert.equal(foundElements.length, 3);
+assert.lengthOf(foundElements.length, 3);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/66984218720e29393b1adad2.md:50]:
```suggestion
assert.lengthOf(foundElements, 3)
```
I'm not leaving anymore suggestions, if you use `assert.legnthOf`, don't use `.length`. Please fix the others

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md [hunk 15]
```diff
@@ -44,7 +44,7 @@ const childrenOf1stSection = [
 const foundElems = childrenOf1stSection.filter((child) => {
   return child.nodeName === 'SECTION';
 });
-assert.equal(foundElems.length, 0);
+assert.isEmpty(foundElems.length);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6698429ba5761b39f8aad814.md:47]:
```suggestion
assert.isEmpty(foundElems);
```

**Review by gikf — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Pull request merged**